examples/Main.hs

```diff
@@ -46,6 +52,53 @@ example2 = plam $ \x -> pmatch x $ \case
   PLeft n -> n + 1
   PRight n -> n - 1
 
+data SampleRecord f = SampleRecord {
+  sampleBool :: f PBool,
+  sampleInt :: f PInteger,
+  sampleString :: f PString}
+
+type instance ScottEncoded SampleRecord a = PBool :--> PInteger :--> PString :--> a
+
+newtype PRecord r s = PRecord{_getRecord :: r (Term s)}
+
+psample :: PRecord SampleRecord s
+psample = PRecord SampleRecord{
+  sampleBool = pcon PFalse,
+  sampleInt = 6,
+  sampleString = "Salut, Monde!"}
+
+instance PlutusType (PRecord SampleRecord) where
+  type PInner (PRecord SampleRecord) a = ScottEncoding SampleRecord a
+  pcon' :: forall a s. PRecord SampleRecord s -> Term s (ScottEncoding SampleRecord a)
+  pcon' (PRecord (SampleRecord b i s)) = plam (\f-> f # b # i # s :: Term s a)
+  pmatch' :: forall s c. (forall b. Term s (ScottEncoding SampleRecord b)) -> (PRecord SampleRecord s -> Term s c) -> Term s c
+  pmatch' r f = r #$ plam $ \b i s -> f (PRecord $ SampleRecord b i s)
+
+sampleRec :: Term s (ScottEncoding SampleRecord t)
+sampleRec = letrec $ const SampleRecord{
+  sampleBool = pcon PTrue,
+  sampleInt = 12,
+  sampleString = "Hello, World!"}
+
+data EvenOdd f = EvenOdd {
+  even :: f (PInteger :--> PBool),
+  odd :: f (PInteger :--> PBool)
+  }
+
+type instance ScottEncoded EvenOdd a = (PInteger :--> PBool) :--> (PInteger :--> PBool) :--> a
```

Comment:
Can you think of any way to avoid this?

Reply:
Do you mean the existence of `ScottEncoded` or the need to declare its instance for every record type? For the latter, I can add a TH splice to automate the job, it would be merged into the existing `$(Rank2.TH.deriveAll SampleRecord)`.

Reply:
Both, preferably, but just the need to declare iet would also be an improvement.